t/serializer_mutable_custom.t

```diff
@@ -0,0 +1,126 @@
+            'application/json'   => 'JSON',
+            'text/other'         => 'Other',
+        } } } };
+    }
+
+    use Ref::Util qw<is_hashref>;
+
+    set serializer => 'Mutable';
+
+    get '/serialize'     => sub { +{ bar => 'baz' } };
+    post '/deserialize'  => sub {
+        return request->data &&
+               is_hashref( request->data ) &&
+               request->data->{bar} ? { bar => request->data->{bar} } : { ret => '?' };
+    };
+}
+
+my $app = MyApp->to_app;
+ok is_coderef($app), 'Got app';
+
+test_psgi $app, sub {
+    my $cb = shift;
+
+    # Configure all test cases
+    my $d = {
+        yaml    => {
+                types       => [ qw(text/x-yaml) ],
+                value       => encode('UTF-8', YAML::Dump({ bar => 'baz' })),
+                last_val    => "---bar:baz",
+            },
+        other    => {
+                types       => [ qw(text/other) ],
+                value       => '{thing}',
+                last_val    => "{thing}",
+            },
+        dumper  => {
+                types       => [ qw(text/x-data-dumper) ],
+                value       => Data::Dumper::Dumper({ bar => 'baz' }),
+                last_val    => "\$VAR1={'bar'=>'baz'};",
+            },
+        json    => {
+                types       => [ qw(text/x-json application/json) ],
+                value       => JSON::MaybeXS::encode_json({ bar => 'baz' }),
+                last_val    => '{"bar":"baz"}',
+            },
+    };
+
+    for my $format (keys %$d) {
+        subtest "Format: $format" => sub {
+
+            my $s = $d->{$format};
+
+            # Response with implicit call to the serializer
+            for my $content_type ( @{ $s->{types} } ) {
+                subtest $content_type => sub {
+                    for my $ct (qw/Content-Type Accept/) {
+
+                        # Test getting the value serialized in the correct format
+                        my $res = $cb->( GET '/serialize', $ct => $content_type );
+
+                        is( $res->code, 200, "status" );
+                        is( $res->content, $s->{value}, "content" );
+                        is(
+                            $res->headers->content_type,
+                            $content_type,
+                            "content-type headers",
+                        );
+                    }
+
+                    # Test sending the value serialized in the correct format
+                    # needs to be de-serialized and returned
+                    my $req = $cb->( POST '/deserialize',
+                                        'Content-Type' => $content_type,
+                                        content        => $s->{value} );
+
+                    my $content = $req->content;
+                    $content =~ s/\s//g;
+                    is( $req->code, 200, "status" );
+                    is( $content, $s->{last_val}, "content" );
+                }
+            }
+        }
+    }
+
+}
```

Comment:
The code repetition here irks me somewhat - feels like the original test script could just create two apps, one that has custom stuff in its mappings and one that doesn't, then loop over them doing all the tests, so no need for a second file of copy & pasted code.  Or, perhaps, even simpler, just provide a custom mapping and test that it all works, then test the default mapping is returned as expected separately - or even don't do that, as we can trust a Moo attribute's default to work properly :)